src/Abstractions/NexusMods.Abstractions.Loadouts/Loadout.cs

```diff
@@ -1,150 +1,167 @@
-    /// <param name="func">Function used to change the details of this mod.</param>
-    /// <returns>A new loadout with the details of a single mod changed.</returns>
-    public Loadout Alter(ModId modId, Func<Mod, Mod?> func)
+    public static IEnumerable<Model> ByName(IDb db, string name)
     {
-        return this with
-        {
-            Mods = Mods.Keep(modId, func)
-        };
+        return db.FindIndexed(name, Name)
+            .Select(db.Get<Model>);
     }
 
     /// <summary>
-    /// Adds an individual mod to this loadout, returning a new loadout.
+    /// Gets all the revisions of a loadout over time
     /// </summary>
-    /// <param name="mod">An individual modification to add to this loadout.</param>
-    /// <returns>The loadout with this modification added.</returns>
-    public Loadout Add(Mod mod)
+    public static IObservable<Model> Revisions(this IConnection conn, LoadoutId id)
     {
-        return this with
-        {
-            Mods = Mods.With(mod.Id, mod)
-        };
+        // All db revisions that contain the loadout id, select the loadout
+        return conn.Revisions
+            .Where(db => db.Datoms(db.BasisTxId).Any(datom => datom.E == id.Value))
+            .StartWith(conn.Db)
+            .Select(db => db.Get<Model>(id.Value));
     }
 
     /// <summary>
-    /// Remove an individual mod from this loadout, returning a new loadout.
+    /// Returns all revisions of a loadouts over time.
     /// </summary>
-    /// <param name="mod"></param>
-    /// <returns></returns>
-    public Loadout Remove(Mod mod)
+    public static IObservable<Model> LoadoutRevisions(this IConnection conn)
     {
-        return this with
-        {
-            Mods = Mods.Without(mod.Id)
-        };
+        // All db revisions that contain a loadout id, select the loadout
+        return conn.Revisions
+            .SelectMany(db => db.Datoms(db.BasisTxId))
+            .Where(d => d.A == Revision)
+            .Select(d => d.E)
+            .StartWith(conn.Db.Find(Revision))
+            .Select(id => conn.Db.Get<Model>(id));
     }
-
-    /// <summary>
-    /// Allows you to change individual files associated with a mod in this collection.
-    /// </summary>
-    /// <param name="func">Function used to change the files of a given mod.</param>
-    /// <returns>A new loadout with the files of mods changed.</returns>
-    public Loadout AlterFiles(Func<AModFile, AModFile?> func)
+    
+    public class Model(ITransaction tx) : Entity(tx)
     {
-        return this with
+        /// <summary>
+        /// The unique identifier for this loadout, casted to a <see cref="LoadoutId"/>.
+        /// </summary>
+        public LoadoutId LoadoutId => LoadoutId.From(Id);
+        
+        /// <summary>
+        /// Gets the loadout id/txid pair for this revision of the loadout.
+        /// </summary>
+        public LoadoutRevisionId LoadoutRevisionId => new(LoadoutId, Db.BasisTxId);
```

Comment:
I'm not sure if I fully understand this. What's the difference between `LoadoutRevisionId` and `ulong Revision`? From my understanding, `LoadoutRevisionId` is a pair of `LoadoutId` and `BasisTxId` (terrible name btw) which points to a specific Loadout with a specific database revision? Meanwhile, `Revision` is just a user-facing monotonic increasing number that essentially represents the number of changes done?

These two things need way better names to prevent confusion. @Al12rs thoughts?

Reply:
Do we need the user facing revision number?